src/harness/security_testcase.py

```diff
@@ -140,6 +154,146 @@ def findIndexMatching(array, regex):
     self.assertTrue(finished_check_idx > 0)
     self.assertTrue(finished_check_idx > cipher_check_idx)
 
+
+
+  def assertVerifyServerCertificateSuceed(self, base_url, ciphers, client_cert, client_key):
+    """ Specific to SCS_20 testcase
+    Checks that the TLS handshake succeed with the given parameters.
+
+    Attempts to establish a TLS session with the given |base_url|, using the
+    given |ciphers| list and the given certificate key pair.
+    Checks that he SAS UUT response must satisfy all of the following conditions:
+    - The SAS UUT agrees to use a cipher specified in the |ciphers| list
+    - The SAS UUT agrees to use TLS Protocol Version 1.2
+    - Valid Finished message is returned by the SAS UUT immediately following
+      the ChangeCipherSpec message
+    """
+    url = urlparse.urlparse('https://' + base_url)
+    client = socket.socket()
+    client.connect((url.hostname, url.port or 443))
+    logging.debug("OPENSSL version: %s" % SSL.SSLeay_version(SSL.SSLEAY_VERSION))
+    logging.debug('TLS handshake: connecting to: %s:%d', url.hostname,
+                  url.port or 443)
+    logging.debug('TLS handshake: ciphers=%s', ':'.join(ciphers))
+    logging.debug('TLS handshake: privatekey_file=%s', client_key)
+    logging.debug('TLS handshake: certificate_file=%s', client_cert)
+    logging.debug('TLS handshake: certificate_chain_file=%s',
+                  self._sas._tls_config.ca_cert)
+
+    ctx = SSL.Context(SSL.TLSv1_2_METHOD)
+    ctx.set_cipher_list(':'.join(ciphers))
+    ctx.use_certificate_file(client_cert)
+    ctx.use_privatekey_file(client_key)
+    ctx.load_verify_locations(self._sas._tls_config.ca_cert)
+
+    client_ssl_informations = []
+    def _InfoCb(conn, where, ok):
+      client_ssl_informations.append(conn.get_state_string())
+      logging.debug('TLS handshake info: %d|%d %s', where, ok, conn.get_state_string())
+      return ok
+    ctx.set_info_callback(_InfoCb)
+
+    
+    default_values = json.load(
+                    open(os.path.join('testcases', 'configs', 'test_WINNF_FT_S_SCS_20', 'default.config')))
+
+    def _ValidateSeverCertificateCb(conn, cert, errnum, depth, ok):
+        certsubject = crypto.X509Name(cert.get_subject())
+        commonname = certsubject.commonName
+        logging.debug('TLS handshake verify: certificate: %s  -> %d', commonname, ok)
+        if(default_values["Subject"] == crypto.X509Name(cert.get_subject()).commonName ):
+          cnotBefore = cert.get_notBefore()
+          cnotAfter = cert.get_notAfter()
+          pubkey = cert.get_pubkey()
+          serial_number = cert.get_serial_number()
+          certsubject = crypto.X509Name(cert.get_issuer())
+          commonname = certsubject.commonName
```

Comment:
duplicate line 202

Reply:
This comment will be addressed as part of   the SCS20 test case submission in a new PR.